lib/blueprinter/view_collection.rb

```diff
@@ -20,7 +21,11 @@ def has_view?(view_name)
     end
 
     def fields_for(view_name)
-      identifier_fields + sortable_fields(view_name).values.sort_by(&:name)
+      sorted_fields = sortable_fields(view_name).values
```

Comment:
This would both be more clear to me and leave ourselves open to change if we assign the variable only once (conditionally) before adding it to `identifier_fields`

ie
```
sorted_fields = if sort_by_definition
  sortable_fields(view_name).values.sort_by(&:name)
else
  sortable_fields(view_name).values
end
identifier_fields + sorted_fields
```

What do you think?

Reply:
Alright I can agree to set the `sorted_fields` variable at the beginning of the method, but I still don't think we should reassign the variable within a condition. If we want to conduct an operation on this collection conditionally, let's be honest about it.

```
sorted_fields = sortable_fields(view_name).values

sorted_fields.sort_by!(:name) unless sort_by_definition

identifier_fields + sorted_fields
```

Alternatively we could treat these two variables as separate things entirely:
```
fields_sorted_by_definition = sortable_fields(view_name).values

fields_sorted_by_name = fields_sorted_by_definition.sort_by(&:name) unless sort_by_definition

identifier_fields + (fields_sorted_by_name || fields_sorted_by_definition)
```

I think the second option is cumbersome, but I would rather have to read more than do a double take at a variable being reassigned in a condition.

Reply:
I agree with your philosophy from the first suggestion and that was my goto option as well, but there ins't a `sort_by!` method available. 😞Hence the current implementation.
From the second suggestion, feels like we're adding one too many variables. Why not reuse the same one? - similar to approach one where we're modifying the original variable itself. They're all bound to the same method anyway and have no usage outside the scope of this method.

Reply:
``` ruby
fields = sortable_fields(view_name).values
sorted_fields = sort_by_definition ? fields.sort_by(:name) : fields
identifier_fields + sorted_fields
```
?

Reply:
Okay, I like your approach and I can make these changes. 

But it's still creating another local variable. Is reassigning to the same local variable considered bad practice? I tend to do that often to avoid having too many variables and think which one refers to what since at the end I'm only bothered about what the method returns.

Reply:
I try to make new variables if possible rather reassigning them. Reassigning just makes it harder to follow in my opinion.